Level-Y/solution/solution.go

```diff
@@ -0,0 +1,92 @@
+// -------------------------------
+// ------------- FIX -------------
+// -------------------------------
+// 1) Remove the email being logged (see line 35)
+// 2) Fix the error message to prevent user enumeration (see line 64)
+// 3) Remove the email and password being logged (see line 71)
+
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"regexp"
+)
+
+var testFakeMockUsers = map[string]string{
+	"user1@example.com": "password12345",
+	"user2@example.com": "B7rx9OkWVdx13$QF6Imq",
+	"user3@example.com": "hoxnNT4g&ER0&9Nz0pLO",
+	"user4@example.com": "Log4Fun",
+}
+
+var reqBody struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+func isValidEmail(email string) bool {
+	// The provided regular expression pattern for email validation by OWASP
+	// https://owasp.org/www-community/OWASP_Validation_Regex_Repository
+	emailPattern := `^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}$`
+	match, err := regexp.MatchString(emailPattern, email)
+	if err != nil {
+		// Fix: Removing the email from the log
+		// log.Printf("Invalid email format: %q", email)
+		log.Printf("Invalid email format")
```

Comment:
I think it makes more sense to print this log in L58, right before the `http.Error` as it happens in `code.go`